From 547691c621d6cf7c9ceab9c90d2097a35e79c1db Mon Sep 17 00:00:00 2001
From: Xavier Hanin <xavier@apache.org>
Date: Wed, 25 Jun 2008 14:40:30 +0000
Subject: [PATCH] handle case where transaction destination directory already
 exists (which makes the commit impossible) (probably related to IVY-733)

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@671574 13f79535-47bb-0310-9956-ffa450edef68
---
 .../plugins/resolver/FileSystemResolver.java  | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/src/java/org/apache/ivy/plugins/resolver/FileSystemResolver.java b/src/java/org/apache/ivy/plugins/resolver/FileSystemResolver.java
index f45f49e52..f35d12939 100644
--- a/src/java/org/apache/ivy/plugins/resolver/FileSystemResolver.java
+++ b/src/java/org/apache/ivy/plugins/resolver/FileSystemResolver.java
@@ -136,6 +136,11 @@ public void commitPublishTransaction() throws IOException {
             if (!isTransactionStarted()) {
                 throw new IllegalStateException("no current transaction!");
             }
+            if (transactionDestDir.exists()) {
+                throw new IOException(
+                    "impossible to commit transaction: transaction destination directory "
+                    + "already exists: " + transactionDestDir);
+            }
             try {
                 getFileRepository().move(transactionTempDir, transactionDestDir);
                 
@@ -172,10 +177,16 @@ public void beginPublishTransaction(
                 unsupportedTransaction("overwrite transaction not supported yet");
             } else {
                 initTransaction(module);
-                Message.verbose(
-                    "\tstarting transaction: publish during transaction will be done in \n\t\t" 
-                    + transactionTempDir 
-                    + "\n\t\tand on commit moved to \n\t\t" + transactionDestDir);
+                if (transactionDestDir.exists()) {
+                    unsupportedTransaction(
+                        "transaction destination directory already exists: " + transactionDestDir);
+                    closeTransaction();
+                } else {
+                    Message.verbose(
+                        "\tstarting transaction: publish during transaction will be done in \n\t\t" 
+                        + transactionTempDir 
+                        + "\n\t\tand on commit moved to \n\t\t" + transactionDestDir);
+                }
             }
         }
     }
